### DISCUSSION TITLE
Errors when routing traffic through Cloudflare

### DISCUSSION THREAD

**Discussion by sourfe** [muraliwebworld/push-notification-plugin-for-wordpress-websites#24] [Q&A]
I get this errors in debug.log when I route the traffic through Cloudflare.

[20-Dec-2024 13:27:15 UTC] PHP Warning:  getimagesize(https://www.xxxxx.se/prodcms/wp-content/uploads/2024/11/SCR-20241115-ggp.png): Failed to open stream: HTTP request failed! HTTP/1.1 403 Forbidden
 in /home/html/xxxxx/public_html/prodcms/wp-content/plugins/push-notification-for-post-and-buddypress/public/service_worker/pnfpb_create_sw_file.php on line 626
[20-Dec-2024 13:27:15 UTC] PHP Fatal error:  Uncaught TypeError: count(): Argument #1 ($value) must be of type Countable|array, bool given in /home/html/xxxxx/public_html/prodcms/wp-content/plugins/push-notification-for-post-and-buddypress/public/service_worker/pnfpb_create_sw_file.php:654
Stack trace:
#0 /home/html/xxxxx/public_html/prodcms/wp-content/plugins/push-notification-for-post-and-buddypress/public/service_worker/pnfpb_create_sw_file.php(62): PNFPB_ic_generate_pwa_manifest_json()
#1 /home/html/xxxxx/public_html/prodcms/wp-includes/class-wp-hook.php(324): PNFPB_icfm_icpush_generate_sw_pwajson()
#2 /home/html/xxxxx/public_html/prodcms/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()
#3 /home/html/xxxxx/public_html/prodcms/wp-includes/plugin.php(565): WP_Hook->do_action()
#4 /home/html/xxxxx/public_html/prodcms/wp-includes/class-wp.php(418): do_action_ref_array()
#5 /home/html/xxxxx/public_html/prodcms/wp-includes/class-wp.php(813): WP->parse_request()
#6 /home/html/xxxxx/public_html/prodcms/wp-includes/functions.php(1336): WP->main()
#7 /home/html/xxxxx/public_html/prodcms/wp-blog-header.php(16): wp()
#8 /home/html/xxxxx/public_html/prodcms/index.php(17): require('...')
#9 {main}
  thrown in /home/html/xxxxx/public_html/prodcms/wp-content/plugins/push-notification-for-post-and-buddypress/public/service_worker/pnfpb_create_sw_file.php on line 654

Web browser console:
Failed to load resource: the server responded with a status of 500 (Internal Server Error) for https://www.xxx.se/pnfpbmanifest.json

**Comment by muraliwebworld** [comment]
If this file(screenshot file) is
https://www.xxxxx.se/prodcms/wp-content/uploads/2024/11/SCR-20241115-ggp.png
cached
in cloudflare then it will be under different domain, you need to supply
correct cloudflare domain cached file name in the settings. This problem
occurred you are using server url for this image in PWA settings but
cloudflare cached this image file in different name, If you are not able to
find correct cloudflare file name then exclude all images url which are
used in pwa settings in cloudflare settings so that it will be taken from
server

**Comment by sourfe** [comment]
I don't really understand what you mean. The request is for the correct domain.

I have bypass on all caching in Cloudflare.

Why is it a problem only for the screenshot?

**Comment by muraliwebworld** [comment]
Due to small incident i got injured  not feeling well in back before some
weeks. I will be back to work after recovering fully i will tesr using
cloudflare and i will let you know. Please always respond with some respect
without knowing actual situation of plugin providers.

**Comment by sourfe** [comment]
It was never my intention to be disrespectful, I was only interested to get some clues about where to start troubleshooting.

Hope you feel better soon, no rush.

**Comment by muraliwebworld** [comment]
Today I routed one of my site in cloudflare free plan, pwa installation with screenshot image is working fine. I have attached screenshots for your reference, (i attached a sample image as screenshot, it will work for any size). Please reattach the attachments, also make sure you have correct redirection (for example https:// to https://www or vice versa , cloudflare ssl with correct settings), please check in your console log if there is due to cors issue image might not be available. I have also attached ping details of my site which shows cloudflare ip address cdn ip address for reference.Still i have some problem in body but i am working only little bit of time. My reponse will be delayed for future responses. 
![Screenshot 2025-01-02 at 6 56 03 PM](https://github.com/user-attachments/assets/0d890503-2038-4f84-a2cc-e86f654b8543)
![Screenshot 2025-01-02 at 6 57 35 PM](https://github.com/user-attachments/assets/13b74558-8b8d-4089-8467-a4e0a87d755c)
![Screenshot 2025-01-02 at 6 56 52 PM](https://github.com/user-attachments/assets/db96607c-06f2-4ec6-afb4-6963e57324f8)
